lib/pages/RidePage.dart

```diff
@@ -163,46 +182,207 @@ class CustomDivider extends StatelessWidget {
+                mainAxisAlignment: MainAxisAlignment.end,
+                children: [
+                  editSingleButton(context),
+                  SizedBox(height: 2),
+                  editAllButton(context),
+                  SizedBox(height: 8),
+                  Padding(
+                    padding: EdgeInsets.only(bottom: 32),
+                    child: ButtonTheme(
+                        height: 50.0,
+                        minWidth: MediaQuery.of(context).size.width * 0.8,
+                        shape: RoundedRectangleBorder(
+                            borderRadius: BorderRadius.all(Radius.circular(10))
+                        ),
+                        child: RaisedButton(
+                          elevation: 3.0,
+                          color: Colors.black,
+                          textColor: Colors.white,
+                          child: Text('Cancel',
+                              style: TextStyle(
+                                  fontSize: 17, fontWeight: FontWeight.bold, color: Colors.red
+                              )
+                          ),
+                          onPressed: () {
+                            Navigator.of(context, rootNavigator: true).pop();
+                            visibilityCallback(true);
+                          },
+                        )
+                    ),
+                  ),
+                ]
+            );
+          }
+      );
+    }
+
+    return Container(
+      width: MediaQuery.of(context).size.width,
+      decoration: BoxDecoration(color: Colors.white, boxShadow: [
+        BoxShadow(
+            spreadRadius: 5,
+            blurRadius: 11,
+            color: Colors.black.withOpacity(0.11))
+      ]),
+      child: Stack(
+        children: <Widget>[
+          Column(
+            mainAxisSize: MainAxisSize.min,
+            children: [
+              ButtonTheme(
                 minWidth: MediaQuery.of(context).size.width * 0.8,
                 height: 50.0,
                 shape: RoundedRectangleBorder(
                     borderRadius: BorderRadius.circular(10)),
-                child: RaisedButton.icon(
-                    onPressed: () {
-                      //TODO: navigate to edit flow
-                    },
-                    elevation: 3.0,
-                    color: Colors.black,
-                    textColor: Colors.white,
-                    icon: Icon(Icons.edit),
-                    label: Text('Edit Ride',
-                        style: TextStyle(
-                            fontSize: 17, fontWeight: FontWeight.bold))),
+                child: Padding(
+                  padding: EdgeInsets.only(top: 18),
+                  child: RaisedButton.icon(
+                      onPressed: () async {
+                        if (ride.parentRide != null) {
+                          showEditDialog();
+                        }
+                        else {
+                          editSingle(context, ride);
```

Comment:
Yep, editSingle is for regular non-recurring rides and for edited instances of rides (don't want to be able to edit all rides from these) and showEditDialog will be for future instances of recurring rides because they'll have the option to edit the single instance or edit all repetitions